devicetypes/smartthings/tint-remote-controller.src/tint-remote-controller.groovy

```diff
@@ -0,0 +1,308 @@
+	return createEvent(name: "level", value: state.currentLevel)
+}
+
+private getButtonEvent(descMap) {
+	def childId = getButtonNumber(descMap.data[3])
+	def description = "Button no. ${childId} was ${value}"
+	def event = createEvent(name: "button", value: "pushed", descriptionText: description, data: [buttonNumber: childId], isStateChange: true)
+	sendEventToChild(childId, event)
+	return event
+}
+
+def on() {
+	state.onOffValue = "on"
+	createEvent(name: "switch", value: "on")
+}
+
+def off() {
+	state.onOffValue = "off"
+	createEvent(name: "switch", value: "off")
+}
+
+def ping() {
+	refresh()
+}
+
+def refresh() {
+	return zigbee.readAttribute(zigbee.ONOFF_CLUSTER, ON_OFF_ATTRIBUTE)
+}
+
+def configure() {
+	def groupsList = [0x4004, 0x4005, 0x4006]
+		groupsList.each {
+			sendHubCommand(addHubToGroup(it))
+		}
+	sendEvent(name: "checkInterval", value: (2 * 60 + 10) * 60, displayed: false, data: [protocol: "zigbee", hubHardwareId: device.hub.hardwareID])
+
+	return zigbee.onOffConfig(5, 3600)
+}
+
+def setColorTemperature(value) {
+	sendEvent(name: "colorTemperature", value: value)
+}
+
+def setLevel(value) {
+	state.currentLevel = value
+	sendEvent(name: "level", value: state.currentLevel)
+}
+
+def setColor(value){
+	sendEvent(name: "hue", value: value.hue)
+	sendEvent(name: "saturation", value: value.saturation)
+}
+
+def installed() {
+	def numberOfButtons = 6
+	sendEvent(name: "numberOfButtons", value: numberOfButtons, displayed: false)
+	addChildButtons(numberOfButtons)
+	state.currentLevel = 100
+	sendEvent(name: "level", value: 100, displayed: false)
+	state.onOffValue = "off"
+	createEvent(name: "switch", value: "off", displayed: false)
+	createEvent(name: "colorTemperature", value: 2700, displayed: false)
+	sendEvent(name: "saturation", value: 100, descriptionText: "Color has changed", displayed: false)
+}
+
+private color_xyYtoHSV(CurrentX, CurrentY) { 
+	// Sources:
+	// https://en.wikipedia.org/wiki/CIE_1931_color_space
+	// http://wiki.nuaj.net/index.php/Color_Transforms
+	// Note: X, Y, Z are tristimulus values, where CurrentX and CurrentY are Zigbee'ss own representation of normalized x and y
+	def x = CurrentX / 65536
+	def y = CurrentY / 65536
+	def z = 1 - x - y
+	def Y = 1
+	def X = (Y / y) * x
+	def Z = (Y / y) * z
+	def RGB = colorXYZtoRGB(X, Y, Z)
+	colorRGBtoHSV(RGB)
+}
+
+private colorXYZtoRGB(X, Y, Z) {
+	def R = X * 3.2406 + Y * -1.5372 + Z * -0.4986
+	def G = X * -0.9689 + Y * 1.8758 + Z * 0.0415
+	def B = X * 0.0557 + Y * -0.2040 + Z * 1.0570
+	R = R > 0.0031308 ? 1.055*Math.pow(R, 1/2.4) - 0.055 : 12.92*R;
+	G = G > 0.0031308 ? 1.055*Math.pow(G, 1/2.4) - 0.055 : 12.92*G;
+	B = B > 0.0031308 ? 1.055*Math.pow(B, 1/2.4) - 0.055 : 12.92*B;
+	clipAndScaleRGBifNedded(R, G, B)
+}
+
+private clipAndScaleRGBifNedded(R, G, B) {
+	def max = [R, G, B].max()
+	R = R / max
+	G = G / max
+	B = B / max
+	R = R < 0 ? 0 : R
+	G = G < 0 ? 0 : G
+	B = B < 0 ? 0 : B
+	[R: R, G: G, B: B]
+}
+
+private colorRGBtoHSV(RGB) {
```

Comment:
We do have a `colorUtil.hexToHsv` already written that takes a 6 digit hex string and outputs HSV. See `rgbw-light`